src/Files.App/Helpers/CloudDrivesDetector.cs

```diff
@@ -92,6 +92,8 @@ private static Task<IEnumerable<ICloudProvider>> DetectGenericCloudDrive()
 						"Amazon Drive" => CloudProviders.AmazonDrive,
 						"Nextcloud" => CloudProviders.Nextcloud,
 						"Jottacloud" => CloudProviders.Jottacloud,
+                        "iCloudDrive!S-1-5-21-4064190285-3015957005-1528788272-1001!Personal" => CloudProviders.AppleCloudDrive,
```

Comment:
Mm this looks a lot like a user SID. Are you sure is the same for everyone? To check: what does this cmd command print? `whoami /user`.

Reply:
```
❯ whoami /user

USER INFORMATION
----------------                                                                                                        
User Name      SID
============== ==============================================
ethan-pc\ethan S-1-5-21-4064190285-3015957005-1528788272-1001
```
i hate everything.